logr.go

```diff
@@ -177,7 +177,7 @@ limitations under the License.
 //   // It does nothing for loggers where the sink doesn't support that parameter.
 //   func WithFoobar(log logr.Logger, foobar int) logr.Logger {
 //      if foobarLogSink, ok := log.Sink(FoobarSink); ok {
-//         log.Sink = foobarLogSink.WithFooBar(foobar)
+//         log.SetSink(foobarLogSink.WithFooBar(foobar))
```

Comment:
Note that LogSink implementations might do this incorrectly and directly set `log.Sink` here.

My preference is to undo the exporting of that field and instead add a `GetSink` method that complements the `SetSink` method from this PR.

Reply:
Agree